### PR TITLE
New package: py-pyusb

### PR DESCRIPTION
**Pull request by healther**
Part of our recent merge and upstream effort.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyusb/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class PyPyusb(PythonPackage):
+    """Easy USB access from Python."""
+
+    homepage = "https://github.com/pyusb/pyusb"
+    pypi     = "pyusb/pyusb-1.0.2.tar.gz"
+
+    version('1.0.2', sha256='4e9b72cc4a4205ca64fbf1f3fff39a335512166c151ad103e55c8223ac147362')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('libusb', type=('build', 'run'))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-pyusb/package.py]: Wouldn't:
```suggestion
    depends_on('libusb')
```
fit better?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyusb/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-pyusb/package.py]: It looks for `libusb` at runtime. I think the dependencies are correct as is

https://github.com/pyusb/pyusb/blob/master/usb/libloader.py

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-pyusb/package.py]: I think this is still `link` per our semantics. `run` is meant for runtime dependencies, like system calls to other tools, that have a separate process space and thus don't need strict unification of symbols. An `ldloaded` library is still `link`. Checking with @becker33 and @tgamblin if I'm saying the correct thing here.

Comment (healther) [reply var/spack/repos/builtin/packages/py-pyusb/package.py]: 👍 I assumed that `link` does not affect the run environment and everything needs to be "compiled into the binary" (for values of compiled and binary). We wouldn't see that distinction as we are (somewhat manually) provide an environment with all (build, link and run) dependencies

File: var/spack/repos/builtin/packages/py-pyusb/package.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class PyPyusb(PythonPackage):
+    """Easy USB access from Python."""
+
+    homepage = "https://github.com/pyusb/pyusb"
+    pypi     = "pyusb/pyusb-1.0.2.tar.gz"
+
+    version('1.0.2', sha256='4e9b72cc4a4205ca64fbf1f3fff39a335512166c151ad103e55c8223ac147362')
+
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyusb/package.py:17]: Missing build dep on `py-setuptools-scm`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyusb/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-pyusb/package.py]: Done

File: var/spack/repos/builtin/packages/py-pyusb/package.py (hunk 2 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-pyusb/package.py:17]: Done, also for good measure I added the scm restriction

File: var/spack/repos/builtin/packages/py-pyusb/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on('py-setuptools-scm@:2.0', type='build', when='^py-setuptools@:12.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyusb/package.py]:
```suggestion
    depends_on('py-setuptools-scm@:1.999', type='build', when='^py-setuptools@:11.999')
```
Version ranges are inclusive

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**